spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/inbound/AbstractHazelcastMessageProducer.java

```diff
@@ -186,6 +193,31 @@ protected void processEvent(EventObject event) {
 			}
 		}
 
+		@SuppressWarnings({ "rawtypes", "unchecked" })
+		@Override
+		protected Message<?> toMessage(EventObject event) {
+			EntryEventMessagePayload<?, ?> messagePayload = null;
+
+			final Map<String, Object> headers = new HashMap<String, Object>();
+			headers.put(HazelcastHeaders.EVENT, ((AbstractIMapEvent) event).getEventType());
+			headers.put(HazelcastHeaders.MEMBER, ((AbstractIMapEvent) event).getMember());
+			headers.put(HazelcastHeaders.NAME, ((AbstractIMapEvent) event).getName());
+
+			if (event instanceof EntryEvent) {
+				Assert.notNull(((EntryEvent) event).getKey(), "key must not be null");
+				messagePayload = new EntryEventMessagePayload(
+						((EntryEvent) event).getKey(), ((EntryEvent) event).getValue(),
+						((EntryEvent) event).getOldValue());
+			}
+			else if (event instanceof MapEvent) {
+				messagePayload = new EntryEventMessagePayload(((MapEvent) event).getNumberOfEntriesAffected());
+			} else {
+				throw new IllegalStateException("Invalid event is received. Event : " + event);
```

Comment:
How does Hazelcast react to this exception?
It is the fact that this `toMessage` is called from the one of those Hazelcast listeners, but it isn't clear to me what is the exception handler engine.
I wonder if we should send it to the `errorChannel`...